packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -234,31 +234,54 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
       PaidStreamingExtensionEvents.REQUEST,
       async (index: number, size: number, response: (allow: boolean) => void) => {
         const reqPrice = bigNumberify(size).mul(WEI_PER_BYTE);
-        const {seedingChannelId, peerAccount: peer, isForceChoking} = wire.paidStreamingExtension;
-        const knownPeer = this.peersList[torrent.infoHash][seedingChannelId];
 
-        if (!knownPeer || !seedingChannelId) {
-          await this.createPaymentChannel(torrent, wire);
-          log.info(`${peer} >> REQUEST BLOCKED (NEW WIRE): ${index}`);
+        const {peerAccount: peer, isForceChoking, seedingChannelId} = wire.paidStreamingExtension;
+
+        if (!seedingChannelId) {
+          // Check to see if we have a running channel with this peer..
+          wire.paidStreamingExtension.seedingChannelId = _.findKey(
+            this.channelsByInfoHash[torrent.infoHash],
+            {
+              id: peer
+            }
+          );
```

Comment:
This needs to be modified such that we don't hook into channels that aren't `running`.

Reply:
8644bcb